### DISCUSSION TITLE
[ANNOUNCEMENT] Response Regarding Google PlayStore App Updates

### DISCUSSION THREAD

**Discussion by agnostic-apollo** [termux/termux-app#4000] [Announcements]
A [`termux-app` `v0.120`](https://play.google.com/store/apps/details?id=com.termux) release was made on Google PlayStore on 2024-06-04 by @fornwall, along with some Termux plugin app releases. These were not done by the current Termux core team. They are only available for Android `>= 11` devices.

The sources for the PlayStore app releases are available at the [`termux-play-store/termux-apps`](https://github.com/termux-play-store/termux-apps) and [`termux-play-store/termux-packages`](https://github.com/termux-play-store/termux-packages) repositories in the [`termux-play-store`](https://github.com/termux-play-store) GitHub organization owned by @fornwall. It is using the `system_linker_exec` solution ([1](https://github.com/fornwall/termux-exec), [2](https://github.com/termux/termux-exec/pull/24#issuecomment-1873059748), [3](https://github.com/agnostic-apollo/Android-Docs/blob/master/site/pages/en/projects/docs/apps/processes/app-data-file-execute-restrictions.md#system-linker-exec)) to bypass [Android `>= 10` exec restrictions](https://github.com/agnostic-apollo/Android-Docs/blob/master/site/pages/en/projects/docs/apps/processes/app-data-file-execute-restrictions.md). Users who wish to install from PlayStore should know that the PlayStore release has [changes reverted](https://github.com/termux-play-store/termux-apps/commit/aa480474b39a6bd4515cebcf72d3ce07d854f23c) for the last `10` Termux app releases ([`v0.109` to `v0.118.0`](https://github.com/termux/termux-app/releases)), and so features, bug and security fixes done in them are not available. **Basically, PlayStore `termux-app` `v0.120` release is functionally equivalent to `v0.108`.** The app also has [reduced permissions](https://github.com/termux-play-store/termux-apps/blob/main/termux-app/src/main/AndroidManifest.xml). It also has certain features removed like the [`RUN_COMMAND` intent](https://github.com/termux/termux-app/wiki/RUN_COMMAND-Intent). Additionally, there are still issues related to `system_linker_exec` solution that are yet to be fixed and the solution also violates [PlayStore policies](https://github.com/agnostic-apollo/Android-Docs/blob/master/site/pages/en/projects/docs/apps/processes/app-data-file-execute-restrictions.md#google-playstore-policies), so the PlayStore release may be taken down in future. So only install from PlayStore at your own risk and if you know what you are doing.

As the PlayStore release has diverged from the [`termux/termux-app` `master` branch](https://github.com/termux/termux-app) on GitHub, it should be considered an **unofficial/forked release source** and its issues should be reported at [`termux-play-store/termux-issues`](https://github.com/termux-play-store/termux-issues).

The only **official release sources** to install Termux from are still F-Droid and GitHub as listed in our [installation docs](https://github.com/termux/termux-app#installation) and Google PlayStore releases [remain deprecated](https://github.com/termux/termux-app#google-play-store-deprecated) at this time.

If you are currently using Termux apps installed from F-Droid and GitHub, then trying to update the apps from PlayStore will fail as the app signing keys are different as documented in the installation docs. If you had installed Termux app from PlayStore before the `v0.120` release, updating will still fail as [`sharedUserId`](https://developer.android.com/guide/topics/manifest/manifest-element#uid) has been removed. To install the PlayStore releases, you need to uninstall the existing Termux app and all its plugin apps first. To prevent PlayStore from attempting to download and then fail to install the PlayStore releases over existing installations, you can open the Termux apps pages on PlayStore and then click on the `3` dots options button in the top right and then disable the `Enable auto update` toggle, however, the Termux apps updates will still show in the PlayStore app updates list.

**Comment by luisdavim** [comment]
A bit crazy that the release was made from an older version and not the latest master...

Can a new release be cut for Github and F-Droid with a higher version so the Play store doesn't try to update what we have installed?

**Reply by lssong99** [reply]
This is the way

**Reply by masterflitzer** [reply]
instead of fdroid version unnecessarily getting bumped, @fornwall should've just used a lower version, this experimental fork seems to be based off 0.108, so why not just make the version 0.108.1, 0.108.2 etc.? there was no reason to use a version number that disrupts everyone, the gplay version isn't ready so shouldn't not be offered to everyone

**Reply by fornwall** [reply]
@masterflitzer Indeed, it was a mistake to use a higher version code, so that Google Play suggests updating away from F-Droid. My mistake - sorry about that :flushed:.

It's not possible to revert now as version code numbers can only go forward, but instead the F-Droid build will have it's version code bumped and released shortly - track https://github.com/termux/termux-app/pull/4005.

> this experimental fork seems to be based off 0.108

That's not the case - while certain features are not yet available (or work differently) in the Google Play build due Google Play policy requirements (or not just being done yet), things are continuously being fixed and implemented as feedback from the review process is adapted to.

**Reply by dilnix** [reply]
I would rather recommend asking Fredrik about renaming the app in the Play Store, or, if it's applicable to change the ID by which the stores define the apps as the same.

**Comment by SaeedDev94** [comment]
IMO play store variant should release with different package name!  
I don't like to see always there is an app update and I can't update due to different signing key 🙄

**Reply by anomalouscowherd** [reply]
Yes.  It would be good if Fornwall was to rename his project but he is unlikely to do that.  I think that this project ( which doesn't have a unique name ) should rename itself to TERMUX 2.0 or something.

**Reply by jddecker** [reply]
Or if the name of this project could be something like Termux-Core or Termux-Origin. It would make it easier to recommend it to people and not have them grab the Play Store version.

**Reply by SaeedDev94** [reply]
Another solution is that termux make reproducible build, so all GitHub, F-Droid and Play Store will be same and this is so much better

**Reply by jacobh12349** [reply]
Or if only the Google Play Store would let you ignore/block updates for a specific app..

**Reply by dgw** [reply]
> Or if only the Google Play Store would let you ignore/block updates for a specific app..

Google changed it relatively recently. It used to ignore updates to apps that weren't installed from Play. Now it shows them, tries to auto update unless disabled, and if course fails to update.

**Reply by mcurly** [reply]
Which in turn from my standpoint is another abuse from Google themselves

**Comment by vvb2060** [comment]
Is that fake software on Google Play Store or is there a split in your team? It's a mess right now and I want to know if app on Play Store can be trusted.

**Reply by twaik** [reply]
Did you read the post?

**Reply by vvb2060** [reply]
Sure, but the post doesn't explain the key point. If termux on play is not trustworthy, why not report the fake software to Google?

**Reply by twaik** [reply]
It is not fake. It is fornwall experiment for coming back to google play core team was not aware of. 
![image](https://github.com/termux/termux-app/assets/9674930/8562c613-2ea3-4cd3-9427-3c18133825ec)

**Reply by matrix0027** [reply]
Does the core team exist anymore when there hasn't been a single update to the F-droid version in years?

**Reply by robertkirkman** [reply]
matrix0027 actually there are multiple updates on the f droid version since you said that

**Comment by yujincheng08** [comment]
Maybe this is helpful to make the play version more useful: https://github.com/termux/termux-exec/pull/24#issuecomment-2156044176

**Comment by Rudxain** [comment]
I have mixed feelings about this. I agree with the ends but **definitely not** the means. I don't like this experiment, unless @fornwall can come up with a perfectly-reasonable justification. Maybe I'm missing some common context? It seems many people agree with this, but I have no idea why.

BTW, don't get me wrong! I'm actually mad at Google, because someone from the Termux team felt the need to do this. W^X should be opt-out, with some sort of ADB command

**Reply by anomalouscowherd** [reply]
Isn't it obvious?  The original termux was monetized. The add-ons where all pay for.  He wants to monetize termux again.  Probably with ads this time.  


FYI:  Fornwall isn't from the "termux team".  He dropped out years ago.

**Reply by ravener** [reply]
> Isn't it obvious? The original termux was monetized. The add-ons where all pay for. He wants to monetize termux again. Probably with ads this time.
> 
> FYI: Fornwall isn't from the "termux team". He dropped out years ago.

That's pretty rude, the plugins were optional and supported the developer but if you wanted it for free it was always available on F-Droid. Fornwall is still a core team member he was just inactive for a while and it's good to see him back and trying to solve the long-standing exec issues.

I agree that this experiment happened out of the blue but give it some time and we'll see where it takes us.

**Reply by jlschuncke** [reply]
> should be opt-out, with some sort of ADB command

I used the venerable Titanium Backup's ability to detach a registered app from the Play Store to opt out.

**Reply by Rudxain** [reply]
> Titanium Backup

AFAIK, that requires unlocking the boot-loader (unless there's a root exploit). Some devices don't even have unlockable BLs, but every "Android™ Certified" device must support ADB, so ADB is more accessible.

But TB seems awesome anyways! Thanks for the recommendation. I might try it when I unlock my BL

**Reply by jlschuncke** [reply]
> But TB seems awesome anyways! Thanks for the recommendation. I might try it when I unlock my BL

Actually, don't bother. It looks like the Play store relinks the app at some point after I detach it using TB, so it's temporary only. (For instance, to get past the pending but impossible update to update other apps, but you can work around that by updating one app at a time.)

**Reply by cameronkelley** [reply]
> Actually, don't bother. It looks like the Play store relinks the app at some point after I detach it using TB, so it's temporary only. (For instance, to get past the pending but impossible update to update other apps, but you can work around that by updating one app at a time.)

If you're rooted and using LSposed, you can hide the app from the play store with UpdateBlocker. Just added termux to my block list and the play store isn't showing the update anymore.

https://xdaforums.com/t/mod-xposed-4-1-update-locker-stop-update-your-apps.4433253/

**Reply by diodasoul** [reply]
This Xposed is great. There are other ways to ignore apps from the Play store, such as using the Xposed module Hide My Applist or the Magisk module zygisk-detach, but this seems to work as another option.

**Comment by nightshade427** [comment]
The play store update makes it so I can't check for other updates from play store. It shows termux is in need of update and the check for updates button is gone. I've installed termux from fdroid, and so play store thinks the play store is the newest version. Can we update the fdroid version to match or make its version greater so my play store will no longer list termux as needing updating.

**Reply by TipsyTheCat** [reply]
this, although being a hacky solution, is definitely imo better than what's currently happening. if the termux team doesn't do this or something else to fix this, I might just arbitrarily change the reported version of the apk and share it. I mean the termux app hasn't needed to be updated in 2 years, so it should be fine to set the version to 1.118

**Comment by fornwall** [comment]
> Can we update the fdroid version to match or make its version greater so my play store will no longer list termux as needing updating.

Yes - that will happen shortly (as in hopefully a few days away)!

**Reply by skj-skj** [reply]
suggestion: playstore version series - 0.x.x and F-Droid/Github version series - 1.x.x

**Reply by anomalouscowherd** [reply]
Odd even. Or better yet merge the trees.

**Reply by pmsobrado** [reply]
Hi, any ETA on this? If I'm not mistaken and haven't miss anything, Google Play version is currently 0.127, and F-droid version is 0.118, so Play Store still tries to update...

**Reply by ewt45** [reply]
@pmsobrado you can check the dev channel https://matrix.to/#/#termux_dev:gitter.im
v0.118.1 and v0.119  has been[ build successfully](https://monitor.f-droid.org/builds/running) in f-droid. they should be signed manually and uploaded soon.

**Reply by pmsobrado** [reply]
But arent 0.118.1 and 0.119 still lower that the 0.127 in Play Store?

**Reply by twaik** [reply]
No, they are not. PS version is based on old source code.

**Reply by pmsobrado** [reply]
I know that. I meant, PS will still ask for update, since F-droid version number is still lower.

**Reply by ewt45** [reply]
v0.127 is jsut a string. what really matters is the versionCode in AndroidManifest.xml。
currently PS's versionCode is 127. f-droid v0.118.1 will be 1000. it's higher that 127 so that PS won't ask for update.

**Reply by pmsobrado** [reply]
Ooooooh I didn't know that. I guess a few more days then... thanks for all the explanations!

**Reply by masterflitzer** [reply]
would have been the perfect opportunity to give version 0.118.1 the versionCode 1181 😅

**Reply by agnostic-apollo** [reply]
There wouldn't be enough patch versions if that version code spec is used on a permanent basis.

https://github.com/termux/termux-app/issues/4012

https://github.com/termux/termux-app/commit/7c5992d379f2e2a3bbf43a608147aaef57d0ca04

**Reply by masterflitzer** [reply]
fair, i didn't think about needing more than 9 patch versions

**Reply by agnostic-apollo** [reply]
`v0.118.*` hopefully wouldn't require that many as development will primarily continue on `v0.119.0*` now, but it didn't make sense to use `1181` just for one version.

**Comment by feuerswut** [comment]
an unofficial release on the play store with the same app id... 😂 who got that idea 😂💀
edit: frederik: You could have just titled it 0.109a or 0.110 not to mention the provided update version integer could have been lower.

**Reply by Rudxain** [reply]
maybe it was an accident? 💀 lmao. I mean, there have been [worse accidents](https://about.gitlab.com/blog/2017/02/01/gitlab-dot-com-database-incident)

**Reply by fornwall** [reply]
> frederik: You could have just titled it 0.109a or 0.110 not to mention the provided update version integer could have been lower

Indeed, having a higher versionCode was a mistake 🤦. It will be resolved shortly instead by bumping the version code in a new F-Droid release, so Google Play will no longer try to update from the F-Droid build.

**Reply by jaredcat** [reply]
> shortly

how much longer? its been days. isn't it simply updating the Gradle file?

**Reply by tareksander** [reply]
f-droid checks for updates each day and builds them from source for all updated apps, and publishing is a manual step after that. In my experience it can take up to 2-4 days for the update to be out worst-case.

**Reply by agnostic-apollo** [reply]
It used to take 5-7 days too, their speeds might have increased now. But a release hasn't been made yet to be detected, possibly today/tomorrow depending on how things go.

**Reply by jaredcat** [reply]
I mean on this repo. The version hasn't been changed yet. 
There should be a new release with an apk we can download by now.

**Reply by nightshade427** [reply]
I don't see an update in fdroid as of yet

**Reply by Dhyfer1** [reply]
Stop waiting. Four days ago version 0.127 was released in the Play Store 👍

**Reply by nightshade427** [reply]
Ohh okay, play store one is now the official one?

**Reply by Dhyfer1** [reply]
> Ohh okay, play store one is now the official one? 

I don't know which one is official, but I always download from the Play Store, and the latest version on Play Store has a higher version number than Termux on GitHub or fdroid

**Reply by masterflitzer** [reply]
> Stop waiting. Four days ago version 0.127 was released in the Play Store 👍

> I don't know which one is official, but I always download from the Play Store, and the latest version on Play Store has a higher version number than Termux on GitHub or fdroid

did you even read this thread??? play store version is an experiment to comply with current google guidelines, it doesn't have full functionality and fdroid is recommended over play store, stop saying we should download it, it is EXPERIMENTAL

higher version number means nothing here, as this play store version is basically a fork which just continued versioning while being experimental, which was a big MISTAKE

**Reply by Dhyfer1** [reply]
> did you even read this thread??? play store version is an experiment to comply with current google guidelines, it doesn't have full functionality and fdroid is recommended over play store, stop saying we should download it, it is EXPERIMENTAL
> 
> higher version number means nothing here, as this play store version is basically a fork which just continued versioning while being experimental, which was a big MISTAKE

Apparently you haven't read the 'What's new' in the latest version in the Play Store. This new version is fully functional, just like the previous version 0.126 and 0.122. In my case I have not had problems with these updated versions from the Play Store and Android 13

**Reply by masterflitzer** [reply]
apparently you didn't read the readme, it is explained very clearly that f-droid is the recommended version: https://github.com/termux/termux-app#google-play-store-experimental-branch

the play store version is literally a fork and even there they recommend fdroid over play store version: https://github.com/termux-play-store#current-status-for-users

you can even see that some features are not yet available from an official comment on this thread: https://github.com/termux/termux-app/discussions/4000#discussioncomment-9772894

the play store what's new doesn't say fully functional: 
![image](https://github.com/termux/termux-app/assets/60170948/33c4a496-8110-42d0-a5d9-3db05672b124)

official termux website also only links to github and fdroid (this is not an accident): https://termux.dev

so please stop spreading misinformation that contradicts official announcements from termux team

**Reply by Dhyfer1** [reply]
As I said, in my case it is totally functional for what I use termux, and that is enough 😊

**Reply by masterflitzer** [reply]
> Basically, PlayStore termux-app v0.120 release is functionally equivalent to v0.108. The app also has reduced permissions, including lack of external storage permission currently. It also has certain features removed like the RUN_COMMAND intent. Additionally, there are still issues related to system_linker_exec solution that are yet to be fixed and the solution also violates PlayStore policies, so the PlayStore release may be taken down in future. So only install from PlayStore at your own risk and if you know what you are doing.

additionally the information is literally in the OP, just because you don't use the missing features, it doesn't mean it should be recommended to everyone

also upvoting your own comment won't increase it's validity

**Reply by Dhyfer1** [reply]
As a basic Termux user, this new version on the Play Store is fully functional for what I use, and that's enough 😊

**Reply by masterflitzer** [reply]
why are you repeating yourself? just move on, when you have nothing to say anymore

**Reply by Dhyfer1** [reply]
You're just upset because termux from the Play Store doesn't work for you but it does for me? Maybe it's your version of Android, or have you already tried clearing the cache and data? If not, try reinstalling the app

**Reply by masterflitzer** [reply]
i am not upset and you are acting like a child here, i pointed out your recommendation is wrong and quoted official announcements

stop assuming things and stay with the facts, i didn't claim it was not working for me or in general, i claimed it's not fully functional which is true, it's good they try to make a play store version, but until it's really ready, a recommendation or let alone telling people to migrate is foolish

I won't migrate termux and all of it's contents (like described in readme under backing up termux) just to try out an experimental release that is not recommended, a fix for fdroid getting higher version code is in the making and i am waiting for that so the annoying play store failed to update message stops appearing

all you said was "and that's enough", which is hardly a good argument and shows that you don't use all the features which doesn't qualify you to make good recommendation unlike the termux team, they know what they are doing and their recommendation is the opposite of yours, especially as installing from fdroid repo is not even an inconvenience, quite the opposite, switching over to google play without any benefits makes no sense at all

i won't waste any more time with you and just hope that people take your comment with a grain of salt and make up their own decision

**Reply by Dhyfer1** [reply]
As I said, as a basic user with Termux, this latest version in the Play Store for what I use it for is fully functional, and since Termux complies with the basic things that I do with the app, it is enough 😊

**Reply by anomalouscowherd** [reply]
Say guys,  It's a real blast to read your informed and thoughtful discussion but maybe you could take this private so I don't have to see this this play out over what looks like 15 messages in my inbox so far?  That would be so cool.  

TIA.

**Reply by Dhyfer1** [reply]
@anomalouscowherd For me there is no problem, It's just that the other guy seems to be having a bad day today. 😄

**Reply by Rudxain** [reply]
@masterflitzer (and everyone) please report @Dhyfer1 for trolling

**Comment by agnostic-apollo** [comment]
Before commenting, I suggest users to join the [Termux Dev Matrix Channel](https://matrix.to/#/#termux_dev:gitter.im) ([Gitter](https://gitter.im/termux/dev)) room and read the dev team comments to get info on changes that are planned for current situation and under normal circumstances. Do not comment there unless you are part of the termux team or have something extremely important and relevant to say that is compliant with the room description, otherwise you will be banned.

**Comment by TipsyTheCat** [comment]
and now Google play is trying to downgrade my termux version SMH this shouldn't be released with the same name or even package, it should be it's own distinct package to test separately from the offical fdroid version

**Comment by charnysh** [comment]
Dear developers!

I have a proposal that I think should satisfy everyone:
1. Remove version 0.120 from the Termux update channel on Google Play and leave everything there as it was before version 0.120 was posted;
2. Open a new Termux Beta channel on Google Play and place version 0.120 there under the name 0.120-beta, change the name of the installation package accordingly.

This will provide the following benefits:
1. There will be no confusion with versions, which causes misunderstanding among users and can damage the reputation of the Termux application;
2. It will be possible to post beta versions with experimental functionality on the Beta channel even after merging version 0.120 with the main version;
3. Users will be able to install both versions if they wish: the main, reliable version of Termux to do their work;  Termux Beta version for experiments to give feedback to developers.

I think this is not a bad idea.

**Reply by twaik** [reply]
> 2\. change the name of the installation package accordingly.

No. Package name is hardcoded in all packages, it is not something we can avoid.

**Reply by charnysh** [reply]
@twaik I suggest changing the name of the installation package, for example from termux.com to termux-beta.com, so that you can install the main and beta versions simultaneously as two different applications.

**Reply by twaik** [reply]
Again. It will not work. Packages expecting "com.termux" package name. With no options. They can not be installed to path other than "/data/data/com.termux/files/".

**Reply by charnysh** [reply]
@twaik  Then how did I install from Google Play: Firefox and Firefox Beta, Opera and Opera Beta?

**Reply by twaik** [reply]
They are different applications with different needs. You can not compare the way they are built and distributed with termux.

**Reply by charnysh** [reply]
Ok, sorry, I wanted what was best for the users.  :)

**Reply by twaik** [reply]
I know you wanted to propose something good. But it simply does not work this way.

**Reply by charnysh** [reply]
@twaik  Thanks for the explanation, I think I understand what the problem is for developers.

**Reply by SaeedDev94** [reply]
> I know you wanted to propose something good. But it simply does not work this way.

What about keeping the play store version code lower than github, f-droid ?

**Reply by fornwall** [reply]
For reference, there is follow up information at: https://www.reddit.com/r/termux/comments/1dbujal/information_about_the_termux_build_on_google_play/

**Reply by charnysh** [reply]
@twaik @fornwall One thought haunts me.
If instead of using a hard-coded installation path for the application, a variable would be used, the value of which would be assigned by the installer during the installation of packages.  For security, the value of the variable should not be arbitrary, but correspond to the one specified in the list controlled by the developers.  For example: the values ​​"/data/data/com.termux/files/" and "/data/data/com.termux-beta/files/" are true and accepted, the value "/data/data/com.termux-clone/  files/" or "/data/data/com.termux-fake/files/" are false and are ignored.
I understand that this idea is probably crazy and difficult to implement, but maybe...

**Reply by twaik** [reply]
It does not work this way. You can not use variables inside elf format fields. Making most programs work with variadic prefix is a HUGE work and nobody will do that without real need.

**Reply by charnysh** [reply]
@twaik I understand, thanks for the explanation.

**Comment by jdimpson** [comment]
Given the current situation, which I am sure will  be resolved , but not before several people start criticizing the dev team, I want to say:

Thank you, dev team, for your hard work on termux. I use it every day, and can't imagine not having it. You do great work, and I sincerely appreciate it.

**Reply by nightshade427** [reply]
This project and the devs are amazing

**Comment by touhidurrr** [comment]
By the way, the last termux version on F-Droid was released in 2022, I wonder why hasn’t been any updates since then. One thing @fornwall did good was to make the community active again. Maybe its time to review all issues and pull requests and release a new version to fix this issue?

**Reply by fornwall** [reply]
> By the way, the last termux version was released in 2022, I wonder why hasn’t been any updates since then.

This was due to Google Play policy requirements introduced at that time, which did not allow Termux to be updated on Google Play.

See https://github.com/termux-play-store for more information.

**Reply by touhidurrr** [reply]
> This was due to Google Play policy requirements introduced at that time, which did not allow Termux to be updated on Google Play.

I meant the F-Droid version actually. Edited.

**Reply by luisdavim** [reply]
The community was active, just no recent releases for the app, but packages are added and updated almost every day. There's more to Termux than just the main app.

**Comment by ztest973004** [comment]
Termux should not be installable on android 7 to 10 (Google Play store) from now.

**Reply by fornwall** [reply]
@mindbound Yes, the plan is to expand it to Android 10 as well (need to debug&fix an issue seen there first).

**Reply by mindbound** [reply]
Thank you!

**Comment by mindbound** [comment]
Termux Styling also has a PlayStore update. Could this be fixed/updated as well?

**Comment by charnysh** [comment]
Today F-Droid updated Termux to version 0.118.1.  Google Play no longer offers to update Termux installed from F-Droid.  The problem is solved, thanks to the developers!

**Reply by nightshade427** [reply]
Awesome 🎉

Now just need termux style update to come through fdroid

**Reply by Dhyfer1** [reply]
You're welcome!

**Reply by luisdavim** [reply]
Will there be an update for termux-styling as well?

**Reply by agnostic-apollo** [reply]
styling and boot updates have already been released on github, will take time for fdroid.

**Reply by charnysh** [reply]
I have Termux:Styling version 0.32.0 (04/20/2024) installed from F-Droid.  Google Play does NOT offer to update this version and this version seems to work without errors with Termux version 0.118.1 from F-Droid.

**Reply by tmzullinger** [reply]
I noticed that `termux-widget` hasn't yet been updated similarly, so I took a stab at that (https://github.com/termux/termux-widget/pull/119).  I don't know if that saves enough effort to be helpful or not.  Thanks for the extra effort in cleaning things up @agnostic-apollo.

Is there someone at F-Droid who should be pinged to pick up these changes?  Neither styling nor boot appear to have been built yet.  I'm not familiar with their processes.  I'll give it a few more days before filing a ticket there to _gently_ nudge it along.

**Reply by tareksander** [reply]
F-droid builds apps automatically every night, but builds can take a long time, can only be picked up on the next build cycle and have to be manually signed and released afterwards. A delay of 2-4 days is expected.

**Reply by agnostic-apollo** [reply]
Check https://gitlab.com/fdroid/wiki/-/wikis/FAQ#how-long-does-it-take-for-my-app-to-show-up-on-website-and-client for more info on timelines. Generally, does take 5-7 days.

Especially check the following part.

>Each build cycle starts at a certain fdroiddata commit. At https://monitor.f-droid.org/builds/running you can see the current cycle.

After metadata file has been updated, and the app needs to be built, then if the commit link at the top of the running page under `fdroiddata version` does not have the updated version in the app metadata file in git history of that commit, then it will not get built, once commit updates in later cycles and the newer commit does have the updated metadata file, then app will get built.

See also https://monitor.f-droid.org/builds/build for built apps. Currently, both apps have been built as per running page, but they are not in build page.

`Build Metadata` file master branch link can be found on F-Droid app page.
 
https://f-droid.org/en/packages/com.termux.styling/
https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/com.termux.styling.yml

**Reply by charnysh** [reply]
F-Droid updated my Termux:Styling to version 0.32.1 today.

**Reply by nightshade427** [reply]
Same here, all updated on fdroid and google play is no longer asking for updates. Thanks y'all!

**Reply by colmbuckley** [reply]
What is happening with termux-widget? I had thought that @agnostic-apollo was working on a new release, but it now seems to be closed as "not planned"?

I am really appreciative of all the cleanup effort here, just a little confused?

**Comment by JanuszChmiel** [comment]
Dear Termux developers,
Thank zou that Termux will be available on Ther Google Play store officially. Because for many users it is problem to install app from The unknown sources. In The todays on security focused world.
By The way. I AM running Ther latest available stable Debian by using proot-distro by using The latest Google Play release of Termux with no issues at all including The mate-session environment, The Orca screen reader.
I Am doing my best to compile my favourite Tuner Internet radio player.
The only sad fact is, that if I Am running Termux or Termux-x11 app I can not answer The incoming phone calls by using touch display.
Unfortnately, this error is not only related for Android accessibility suite, but also this error is existing also if I Am running other screen readers for Android.
But I Am doing real thing to overcome this issue.
I will be using my wired handsfree to answer incoming calls for now.
And may be, that some of The elite developers will be able to find out The cause of The strange problem while using screen readers and window for answerring The incoming phone calls.
I Am also suggesting to you to add new discussing topic called tests of packages needed.
Developers of Termux could create this topic to announce which Terrmux specific packages they would like to test on Android 11 devices. Users could participate.
You could also give step by step instructions how to obtain needed logs if something would happen.

**Comment by Ar-019** [comment]
Still i can use adb to install apps.

**Reply by Rudxain** [reply]
[Not for long](https://github.com/melontini/bootloader-unlock-wall-of-shame/blob/7d72aecb5c0d6a61d874e9bf2a03b4c254a029dc/README.md?plain=1#L24)

**Reply by Buggem** [reply]
@Rudxain what you linked has no relevance whatsoever, and is about bootloaders unlocking.

"They've already gone after sideloading" is old news.

**Comment by mcurly** [comment]
Hi everyone!

Presently, I have build version:  `0.118.0+30ebb2d ` installed on my android tablet from which I am writing this.

(I have yet to find out if my termux installed build is the latest one, since I am assuming that it is)...

Why are these things (still?) happening?

<img width="1920" height="1200" alt="Screenshot_20260425-200456" src="https://github.com/user-attachments/assets/97b07e51-d0d7-4a0b-85e3-ea6a92daea9e" />

<img width="1920" height="1200" alt="Screenshot_20260425-195249" src="https://github.com/user-attachments/assets/73b93abe-a661-4f6b-8a19-5c2ce2e14d50" />


<img width="1920" height="1200" alt="Screenshot_20260425-195300" src="https://github.com/user-attachments/assets/0a8beeac-6052-439b-ad96-c3e9296a4575" />

**Thank you!!!**

🔆

**Reply by agnostic-apollo** [reply]
Install Termux app from github releases, the master branch does not have bumped version code.

**Reply by mcurly** [reply]
Hi, @agnostic-apollo !

**Thank you!**

(Wasn't aware that GitHub actions builds could entangle this sort of behavior...)

So thank you very much for pointing me to the solution! 

**Cheers!**


🔆
